src/types/utils.cpp

```diff
@@ -318,6 +318,84 @@ catch (...)
     return std::nullopt;
 }
 
+// Routine Description:
+// - Constructs a til::color value from RGB percentage components.
+// Arguments:
+// - r - The red component of the color (0-100%).
+// - g - The green component of the color (0-100%).
+// - b - The blue component of the color (0-100%).
+// Return Value:
+// - The color defined by the given components.
+til::color Utils::ColorFromRGB100(const int r, const int g, const int b) noexcept
+{
+    // The color class is expecting components in the range 0 to 255,
+    // so we need to scale our percentage values by 255/100. We can
+    // optimise this conversion with a pre-created lookup table.
+    static constexpr auto scale100To255 = [] {
+        std::array<uint8_t, 101> lut{};
+        for (size_t i = 0; i < std::size(lut); i++)
+        {
+            lut.at(i) = gsl::narrow_cast<uint8_t>((i * 255 + 50) / 100);
+        }
+        return lut;
+    }();
+
+    const auto red = til::at(scale100To255, std::min<unsigned>(r, 100u));
+    const auto green = til::at(scale100To255, std::min<unsigned>(g, 100u));
+    const auto blue = til::at(scale100To255, std::min<unsigned>(b, 100u));
+    return { red, green, blue };
+}
+
+// Routine Description:
+// - Constructs a til::color value from HLS components.
+// Arguments:
+// - h - The hue component of the color (0-360°).
+// - l - The luminosity component of the color (0-100%).
+// - s - The saturation component of the color (0-100%).
+// Return Value:
+// - The color defined by the given components.
+til::color Utils::ColorFromHLS(const int h, const int l, const int s) noexcept

```

Comment:
Eventually, I will file a workitem for moving this stuff into til/color.h itself (not necessarily into the `color` class, mind). We have about three copies of the HSL/HLS code, and perhaps we can get it down to 1. :smile:

Reply:
Yeah, I'm totally on board with this. I was initially expecting to put it somewhere related to the `til::color` class - I only went with `Utils` because that's what we were doing for the XTerm color routines. But as you noted below, these conversions are somewhat unique, and would probably need to be named something like `ColorFromDecRGB` and `ColorFromDecHLS`.